python/ucxx/_lib/libucxx.pyx

```diff
@@ -1146,10 +1176,13 @@ cdef class UCXEndpoint():
 
         return UCXRequest(<uintptr_t><void*>&req, self._enable_python_future)
 
-    def tag_send(self, Array arr, size_t tag):
+    def tag_send(self, Array arr, tag: UCXXTagMask):
```

Comment:
```suggestion
    def tag_send(self, Array arr, UCXXTag tag):
```
Then there's no need for the typerror because Cython will automatically put in the type checking for us.

Reply:
Yeah, unfortunately that doesn't work with Cython in a "simple" way:

```
Error compiling Cython file:
------------------------------------------------------------
...
                self._enable_python_future
            )

        return UCXRequest(<uintptr_t><void*>&req, self._enable_python_future)

    def tag_send(self, Array arr, UCXXTag tag):
                                  ^
------------------------------------------------------------

/datasets/pentschev/src/ucxx/python/ucxx/_lib/libucxx.pyx:1179:34: 'UCXXTag' is not a type identifier
```

As far as I could understand, we can't have C++ types in `def`, we could have a Cython value there similar to what we do with `Array`, that would unfortunately increase even more the complexity here, we need both a new `.pxd` file containing the class' member structure and a `.pyx` file containing its methods definitions. I'd rather leave this for a future PR then. If you know of a simpler way then I'm happy to do it here.